python-avd/pyavd/_eos_designs/structured_config/overlay/router_bgp.py

```diff
@@ -182,49 +182,61 @@ def _address_family_ipv4(self: AvdStructuredConfigOverlay) -> dict:
         peer_groups = []
 
         if self.shared_utils.is_wan_router:
-            peer_groups.append({"name": self.shared_utils.bgp_peer_groups["wan_overlay_peers"]["name"], "activate": False})
+            peer_groups.append({"name": self._get_peer_group_name("wan_overlay_peers"), "activate": False})
 
         # TODO: no elif
         elif self.shared_utils.overlay_evpn_vxlan is True:
-            peer_groups.append({"name": self.shared_utils.bgp_peer_groups["evpn_overlay_peers"]["name"], "activate": False})
+            peer_groups.append({"name": self._get_peer_group_name("evpn_overlay_peers"), "activate": False})
 
         if self.shared_utils.overlay_routing_protocol == "ebgp" and (
             self.shared_utils.evpn_gateway_vxlan_l2 is True or self.shared_utils.evpn_gateway_vxlan_l3 is True
         ):
-            peer_groups.append({"name": self.shared_utils.bgp_peer_groups["evpn_overlay_core"]["name"], "activate": False})
+            peer_groups.append({"name": self._get_peer_group_name("evpn_overlay_core"), "activate": False})
 
         if self.shared_utils.overlay_routing_protocol == "ibgp":
             if self.shared_utils.overlay_mpls is True:
-                peer_groups.append({"name": self.shared_utils.bgp_peer_groups["mpls_overlay_peers"]["name"], "activate": False})
+                peer_groups.append({"name": self._get_peer_group_name("mpls_overlay_peers"), "activate": False})
 
             if self._is_mpls_server is True:
-                peer_groups.append({"name": self.shared_utils.bgp_peer_groups["rr_overlay_peers"]["name"], "activate": False})
+                peer_groups.append({"name": self._get_peer_group_name("rr_overlay_peers"), "activate": False})
 
             if self._is_wan_server_with_peers:
-                peer_groups.append({"name": self.shared_utils.bgp_peer_groups["wan_rr_overlay_peers"]["name"], "activate": False})
+                peer_groups.append({"name": self._get_peer_group_name("wan_rr_overlay_peers"), "activate": False})
 
         if self.shared_utils.overlay_ipvpn_gateway is True:
-            peer_groups.append({"name": self.shared_utils.bgp_peer_groups["ipvpn_gateway_peers"]["name"], "activate": False})
+            peer_groups.append({"name": self._get_peer_group_name("ipvpn_gateway_peers"), "activate": False})
 
         return {"peer_groups": peer_groups}
 
+    def _get_peer_group_name(self: AvdStructuredConfigOverlay, key: str) -> str:
+        """Helper to retrieve the configured peer_group name for the given key."""
+        try:
+            return self.shared_utils.bgp_peer_groups[key]["name"]
+        except KeyError as e:
+            msg = f"`bgp_peer_groups` key {key} does not exist."
+            raise AristaAvdError(msg) from e
```

Comment:
This should never happen. We need to improve the `shared_utils.bgp_peer_groups` instead.